### ISSUE TITLE
Test Python 3.6

### ISSUE THREAD

**Issue by mrocklin** [dask/dask#1811]
We should test Dask in anticipation of Python 3.6.  It looks like there are some builds on conda-forge if we want to use travis.ci.  https://github.com/conda-forge/python-feedstock/pull/45

**Comment by pitrou** [comment]
Unfortunately it looks like we are hitting a conda bug:
```
$ conda create -n dask36 python=3.6 -c conda-forge
Fetching package metadata .............
Solving package specifications: ....


UnsatisfiableError: The following specifications were found to be in conflict:
  - python 3.6*
Use "conda info <package>" to see the dependencies for each package.
```

**Comment by pitrou** [comment]
Oh, well, that's because Python 3.6 doesn't exist in the conda-forge repos (yet?).

**Comment by jreback** [comment]
you need something like this:

https://github.com/pandas-dev/pandas/blob/master/ci/install-3.6_DEV.sh

**Comment by pitrou** [comment]
Oh, so dependencies don't have any binary packages for them? Mmh.

**Comment by mrocklin** [comment]
We can also wait if this is a pain to do.  We can probably wait a while after the release if necessary.

**Comment by jreback** [comment]
yeah nothing really setup atm
you have to install it all

**Comment by mrocklin** [comment]
OK, this issue might be premature then

**Comment by pitrou** [comment]
Still, some first results:
* cloudpickle is not 3.6-compatible (changes in the bytecode format):
```
  File "/home/antoine/miniconda3/envs/dask36/lib/python3.6/site-packages/cloudpickle/cloudpickle.py", line 287, in extract_code_globals
    out_names.add(names[oparg])
```
* the same error on numerous dataframe tests (looks like a Pandas issue):
```
  File "pandas/tslib.pyx", line 1516, in pandas.tslib.convert_str_to_tsobject (pandas/tslib.c:26859)
  File "pandas/src/datetime.pxd", line 141, in datetime._string_to_dts (pandas/tslib.c:90460)
SystemError: <class 'str'> returned a result with an error set
```
* another Pandas-related issue:
```
  File "pandas/tslib.pyx", line 1478, in pandas.tslib.convert_to_tsobject (pandas/tslib.c:26440)
TypeError: Cannot convert input to Timestamp
```
* apparently `test_rolling_functions_series` requires Scipy installed and doesn't skip

The rest looks ok (though some parts weren't exercised because of missing dependencies: blosc, etc.).

**Comment by pitrou** [comment]
Due to the cloudpickle issue, it's probably not worth it testing distributed right now.

**Comment by jreback** [comment]
note all of the pandas issues are fixed in master (another reason for 0.19.2)

https://travis-ci.org/pandas-dev/pandas/jobs/178698199

**Comment by pitrou** [comment]
With a fixed cloudpickle, distributed seems fine except for the pandas-related failures above.

**Comment by jcrist** [comment]
Fixed by #1874. Closing.
